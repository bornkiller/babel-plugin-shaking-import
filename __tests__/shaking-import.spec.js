/**
 * @description - sub-module import test suits
 * @author - huang.jian <hjj491229492@hotmail.com>
 */

// package
const path = require('path');
const tester = require('babel-plugin-tester');

// internal
const shakingImportPlugin = require('../src');

tester.default({
  plugin: shakingImportPlugin,
  pluginName: 'babel-plugin-shaking-import',
  snapshot: true,
  tests: [
    /***********************************************************************/
    /* antd suits */
    /***********************************************************************/
    {
      title: 'antd flavor',
      fixture: path.join(__dirname, '__fixtures__', 'antd.js'),
      pluginOptions: {
        libraryName: 'antd',
        libraryDirectory: 'lib',
        libraryStrategy: 'camel2dash',
      },
    },
    {
      title: 'antd flavor',
      fixture: path.join(__dirname, '__fixtures__', 'antd.js'),
      pluginOptions: {
        libraryName: 'antd',
        libraryDirectory: 'lib',
        libraryStrategy: 'camel2dash',
        libraryStyle: true,
      },
    },
    {
      title: 'antd flavor',
      fixture: path.join(__dirname, '__fixtures__', 'antd.js'),
      pluginOptions: {
        libraryName: 'antd',
        libraryDirectory: 'lib',
        libraryStrategy: 'camel2dash',
        libraryStyle: 'css',
      },
    },
    /***********************************************************************/
    /* lodash flavor */
    /***********************************************************************/
    {
      title: 'lodash flavor',
      fixture: path.join(__dirname, '__fixtures__', 'lodash.js'),
      pluginOptions: {
        libraryName: 'lodash',
        libraryDirectory: '.',
      },
    },
    /***********************************************************************/
    /* react-toolbox flavor */
    /***********************************************************************/

    {
      title: 'react-toolbox flavor',
      fixture: path.join(__dirname, '__fixtures__', 'react-toolbox.js'),
      pluginOptions: {
        libraryName: 'react-toolbox',
        libraryDirectory: 'lib',
        libraryStrategy: 'camel2underline',
      },
    },
    /***********************************************************************/
    /* rxjs flavor */
    /***********************************************************************/
    {
      title: 'rxjs flavor',
      fixture: path.join(__dirname, '__fixtures__', 'rxjs.js'),
      pluginOptions: {
        libraryName: 'rxjs',
        libraryDirectory: '.',
        libraryNameImport: true,
      },
    },
    /***********************************************************************/
    /* skip import default declaration */
    /***********************************************************************/
    {
      title: 'skip import default declaration',
      fixture: path.join(__dirname, '__fixtures__', 'skip.js'),
      pluginOptions: {
        libraryName: 'lodash',
        libraryDirectory: '.',
      },
    },
  ],
});
