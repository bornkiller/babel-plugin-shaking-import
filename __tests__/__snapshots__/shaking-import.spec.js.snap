// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-shaking-import should ignore default import declaration 1`] = `
"'use strict';

import _, { forEachRight } from 'lodash';

forEachRight([1, 2], function (value) {
  console.log(value);
});

_.dropWhile([1, 2, 3, 4], function (value) {
  return value <= 3;
});"
`;

exports[`babel-plugin-shaking-import should support adaptor libraryStrategy 1`] = `
"'use strict';

import forEachRight from 'lodash/forEachRight';
import dropWhile from 'lodash/dropWhile';


forEachRight([1, 2], function (value) {
  console.log(value);
});

dropWhile([1, 2, 3, 4], function (value) {
  return value <= 3;
});"
`;

exports[`babel-plugin-shaking-import should support antd camel2dash style cases 1`] = `
"'use strict';

import Button from 'antd/lib/button';
import DatePicker from 'antd/lib/date-picker';
import message from 'antd/lib/message';


message.info('babel-plugin-shaking-import');

ReactDOM.render(React.createElement(
  'div',
  null,
  React.createElement(DatePicker, null),
  React.createElement(
    Button,
    null,
    'babel-plugin-shaking-import'
  )
));"
`;

exports[`babel-plugin-shaking-import should support antd camel2dash style cases 2`] = `
"'use strict';

import 'antd/lib/button/style';
import Button from 'antd/lib/button';
import 'antd/lib/date-picker/style';
import DatePicker from 'antd/lib/date-picker';
import 'antd/lib/message/style';
import message from 'antd/lib/message';


message.info('babel-plugin-shaking-import');

ReactDOM.render(React.createElement(
  'div',
  null,
  React.createElement(DatePicker, null),
  React.createElement(
    Button,
    null,
    'babel-plugin-shaking-import'
  )
));"
`;

exports[`babel-plugin-shaking-import should support antd camel2dash style cases 3`] = `
"'use strict';

import 'antd/lib/button/style/css';
import Button from 'antd/lib/button';
import 'antd/lib/date-picker/style/css';
import DatePicker from 'antd/lib/date-picker';
import 'antd/lib/message/style/css';
import message from 'antd/lib/message';


message.info('babel-plugin-shaking-import');

ReactDOM.render(React.createElement(
  'div',
  null,
  React.createElement(DatePicker, null),
  React.createElement(
    Button,
    null,
    'babel-plugin-shaking-import'
  )
));"
`;

exports[`babel-plugin-shaking-import should support antd camel2underline style cases 1`] = `
"'use strict';

import TimePicker from 'react-toolbox/lib/time_picker';


ReactDOM.render(React.createElement(
  'div',
  null,
  React.createElement(TimePicker, null)
));"
`;

exports[`babel-plugin-shaking-import should support array state options 1`] = `
"'use strict';

import forEachRight from 'lodash/forEachRight';
import dropWhile from 'lodash/dropWhile';
import 'antd/lib/button/style/css';
import Button from 'antd/lib/button';
import 'antd/lib/date-picker/style/css';
import DatePicker from 'antd/lib/date-picker';
import 'antd/lib/message/style/css';
import message from 'antd/lib/message';


forEachRight([1, 2], function (value) {
  console.log(value);
});

dropWhile([1, 2, 3, 4], function (value) {
  return value <= 3;
});

message.info('babel-plugin-shaking-import');

ReactDOM.render(React.createElement(
  'div',
  null,
  React.createElement(DatePicker, null),
  React.createElement(
    Button,
    null,
    'babel-plugin-shaking-import'
  )
));"
`;

exports[`babel-plugin-shaking-import should support libraryOverride cases 1`] = `
"'use strict';

import forEachRight from 'lodash/forEachRight';
import dropWhile from 'lodash/dropWhile';


forEachRight([1, 2], function (value) {
  console.log(value);
});

dropWhile([1, 2, 3, 4], function (value) {
  return value <= 3;
});"
`;

exports[`babel-plugin-shaking-import should support libraryOverride cases 2`] = `
"'use strict';

import forEachRight from 'lodash-es/forEachRight';
import dropWhile from 'lodash-es/dropWhile';


forEachRight([1, 2], function (value) {
  console.log(value);
});

dropWhile([1, 2, 3, 4], function (value) {
  return value <= 3;
});"
`;

exports[`babel-plugin-shaking-import should support lodash camel2camel style cases 1`] = `
"'use strict';

import forEachRight from 'lodash/forEachRight';
import dropWhile from 'lodash/dropWhile';


forEachRight([1, 2], function (value) {
  console.log(value);
});

dropWhile([1, 2, 3, 4], function (value) {
  return value <= 3;
});"
`;

exports[`babel-plugin-shaking-import should support rxjs preserve cases 1`] = `
"import { Observable } from 'rxjs/Observable';
import { Subject } from 'rxjs/Subject';


// Normal import
import 'rxjs/observable/interval';
import 'rxjs/observable/zip';

const trigger$ = Reflect.construct(Subject, []);
const timer$ = Observable.interval(1000);

Observable.zip(trigger$, timer$).subscribe(value => {
  console.log(value);
});"
`;
